http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -224,6 +225,10 @@ private ChannelPool createChannelPool(Bootstrap bootstrap, ChannelPipelineInitia
         // from the underlying pool, the channel is closed and released.
         channelPool = new CancellableAcquireChannelPool(bootstrap.config().group().next(), channelPool);
 
+        // Wrap the channel pool to guarantee all channels checked out are healthy, and all unhealthy channels checked in are
+        // closed.
+        channelPool = new HealthCheckedChannelPool(bootstrap.config().group(), configuration, channelPool);
```

Comment:
We should switch with `CancellableAcquireChannelPool`

Reply:
Fixed.